src/cogent3/app/composable.py

```diff
@@ -848,15 +850,17 @@ def _disconnect(self):
 
 
 def _add(self, other):
-    if other.input is not None:
+    if hasattr(other, "input") and other.input is not None:
```

Comment:
Remove this `hasattr` check. Instead add a new conditional before this
```python
if getattr(other, `app_type`, None) not in {WRITER, LOADER, GENERIC}:
    raise TypeError(f"{other!r} is not composable"
```

Reply:
Why for the first condition you used getattr, but for the next, directly .app_type?